plugins/modules/discovery_rule.py

```diff
@@ -0,0 +1,149 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: discovery_rule
+version_added: x.x.x
+short_description: Manage Host Discovery Rules
+description:
+  - Manage Host Discovery Rules
+author:
+  - "Jeffrey van Pelt (@Thulium-Drake)"
+options:
+  name:
+    description:
+      - Name of the Discovery Rule
+    required: True
+    type: str
+  search:
+    description:
+      - Expression to match newly discovered hosts with
+    required: True
+    type: str
+  hostgroup:
+    description:
+      - Hostgroup to assign hosts to
+    required: True
+    type: str
+  hostname:
+    description:
+      - Hostname to assign to discovered host(s)
+      - When matching multiple hosts, must provide unique hostnames for each of the discovered hosts
+    type: str
+  enabled:
+    description:
+      - Enable or disable the rule
+    type: bool
+  priority:
+    description:
+      - Priority of the rule
+    type: int
+  max_count:
+    description:
+      - Maximum amount of hosts to provision with the rule
+      - 0 means no limit
+    type: int
+extends_documentation_fragment:
+  locations:
+    description:
+      - List of locations the discovery rule should be assigned to
+    type: list
+    elements: str
+  organizations:
+    description:
+      - List of organizations the discovery rule should be assigned to
+    type: list
+    elements: str
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.entity_state
+  - theforeman.foreman.foreman.nested_parameters
+'''
+
+EXAMPLES = '''
+- name: 'Ensure Discovery Rule'
+    theforeman.foreman.discovery_rule:
+      username: 'admin'
+      password: 'secret_password'
+      server_url: 'https://foreman.example.com'
+      name: 'my-first-disco'
+      search: 'mac = bb:bb:bb:bb:bb:bb'
+      hostgroup: 'RedHat7-Base'
+      hostname: 'servera'
+      max_count: 1
+      organizations:
+        - 'MyOrg'
+      locations:
+        - 'DC1'
+
+- name: 'Remove Discovery Rule'
+  theforeman.foreman.discovery_rule:
+    username: 'admin'
+    password: 'secret_password'
+    server_url: 'https://foreman.example.com'
+    name: 'my-first-disco'
+    state: 'absent'
+'''
+
+RETURN = '''
+entity:
+  description: Final state of the affected entities grouped by their type.
+  returned: success
+  type: dict
+  contains:
+    discovery_rules:
+      description: List of discovery rules.
+      type: list
+      elements: dict
+'''
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import ForemanEntityAnsibleModule, NestedParametersMixin
+
+
+class ForemanDiscoveryRuleModule(NestedParametersMixin, ForemanEntityAnsibleModule):
```

Comment:
If you use `ForemanTaxonomicEntityAnsibleModule` as a base, you won't need to define orgs/locs as params yourself

Reply:
Also, I don't think you need NestedParameters, disco rules have no params.

Reply:
I stole this from ```location.py``` and edited it to make it work :-)

Thanks for the feedback! I've updated the module!